deploy/1_deploy-ens.ts

```diff
@@ -0,0 +1,140 @@
+import { deploy } from '../utils/deploy-tools';
+
+module.exports = async function () {
+    console.log('Deploying and configuring ENS..');
+    const chainId = (await ethers.provider.getNetwork()).chainId;
+    if (chainId < 1000) {
+        // skip ENS setup for mainnet and testnet
+        return;
+    }
+    const [owner] = await hre.ethers.getSigners();
+    const erc1538ProxyAddress = (await deployments.get('ERC1538Proxy')).address;
+    const iexecAccessorsInstance = IexecAccessors__factory.connect(erc1538ProxyAddress, owner);
+    const appRegistryAddress = await iexecAccessorsInstance.appregistry();
+    const datasetRegistryAddress = await iexecAccessorsInstance.datasetregistry();
+    const workerpoolRegistryAddress = await iexecAccessorsInstance.workerpoolregistry();
+    const ens = (await deploy(new ENSRegistry__factory(), owner, [])) as ENS;
+    const resolver = (await deploy(new PublicResolver__factory(), owner, [
+        ens.address,
+    ])) as PublicResolver;
+    const reverseRegistrar = (await deploy(new ReverseRegistrar__factory(), owner, [
+        ens.address,
+        resolver.address,
+    ])) as ReverseRegistrar;
+    const registrars: { [name: string]: FIFSRegistrar } = {};
+    // root registrar
+    await registerDomain('');
+    await registrars[''].register(labelhash('reverse'), owner.address).then((tx) => tx.wait());
+    await ens
+        .setSubnodeOwner(
+            ethers.utils.namehash('reverse'),
+            labelhash('addr'),
+            reverseRegistrar.address,
+        )
+        .then((tx) => tx.wait());
+    await registerDomain('eth');
+    await registerDomain('iexec', 'eth');
+    await registerDomain('v5', 'iexec.eth');
+    await registerDomain('users', 'iexec.eth');
+    await registerDomain('apps', 'iexec.eth');
+    await registerDomain('datasets', 'iexec.eth');
+    await registerDomain('pools', 'iexec.eth');
+    await registerAddress('admin', 'iexec.eth', owner.address);
+    await registerAddress('rlc', 'iexec.eth', await iexecAccessorsInstance.token());
+    await registerAddress('core', 'v5.iexec.eth', erc1538ProxyAddress);
+    await registerAddress('apps', 'v5.iexec.eth', appRegistryAddress);
+    await registerAddress('datasets', 'v5.iexec.eth', datasetRegistryAddress);
+    await registerAddress('workerpools', 'v5.iexec.eth', workerpoolRegistryAddress);
+    await reverseRegistrar.setName('admin.iexec.eth').then((tx) => tx.wait());
+    await setReverseName(erc1538ProxyAddress, 'core.v5.iexec.eth');
+    await setReverseName(appRegistryAddress, 'apps.v5.iexec.eth');
+    await setReverseName(datasetRegistryAddress, 'datasets.v5.iexec.eth');
+    await setReverseName(workerpoolRegistryAddress, 'workerpools.v5.iexec.eth');
+
+    /**
+     * Register domain on ENS.
+     */
+    async function registerDomain(label: string, domain: string = '') {
+        const name = domain ? `${label}.${domain}` : `${label}`;
+        const labelHash = label ? labelhash(label) : '0x0';
```

Comment:
Why not?
```suggestion
        const labelHash = label ? labelhash(label) : ethers.constants.HashZero;
```

Reply:
Both works, I can update yes, see https://github.com/iExecBlockchainComputing/PoCo/pull/93/commits/33ec55da8ffbce25a76cc9670b0ad87632195a86 !